Feature/edit stage

wwqrd opened the pull request:
This implements a single stage editor for the "information interface".

Features include:
- Creating a protocol ' project'
  - This should create a '.canvas' folder, with a protocol.json and an empty 'assets/' directory
- Loading that protocol 'project'
  - Should load up the protocol and display the protocol editor
- Saving the loaded protocol project
  - Should update protocol.json in the currently loaded protocol
- Editing an instructions stage, with asset importing.
  - when adding an asset it should copy it to the 'assets/' folder and display a preview.

## Known issues
- UI assets aren't showing up correctly, fonts, and images from public
- There's no way to return to the homepage
- Some visual glitches with the guidance when dragging and dropping items

wwqrd commented on src/components/StageEditor/sections/ContentItems.js:
```diff
@@ -0,0 +1,129 @@
+            }
+
+            <div className="stage-editor-section-content-items__controls">
+              <AddButton onClick={() => this.createNewItem('text')} type="text">Copy</AddButton>
```
"Copy" the Noun :) I'll rename it.

jthrilly replied:
You mean text. I get it now. Doh.

wwqrd commented:
I've changed the file names and 'copy' button.

> - add a 'label' property which is a short description for the stage to be used in the menu.
> - rename contentItems to be items

I've implemented the above in: https://github.com/codaco/Architect/pull/24

wwqrd commented:
To make tests pass there are some issues with the colorDictionary that I need to look into. Strictly speaking it's out the scope of this task so I'll make a new issue for it.

wwqrd commented:
For some reason this won't build on travis, but it builds locally. I'm going to go ahead and merge this and revisit that later.
